.ibm/pipelines/utils.sh

```diff
@@ -1098,3 +1098,25 @@ detect_ocp_and_set_env_var() {
   fi
   echo IS_OPENSHIFT: "${IS_OPENSHIFT}"
 }
+
+# Helper function for cross-platform sed
+sed_inplace() {
+  if [[ "$OSTYPE" == "darwin"* ]]; then
+    # macOS
+    sed -i '' "$@"
+  else
+    # Linux
+    sed -i "$@"
+  fi
+}
+
+# Helper function for case conversion
+to_lowercase() {
+  if [[ "$OSTYPE" == "darwin"* ]]; then
+    # macOS - using tr
+    echo "$1" | tr '[:upper:]' '[:lower:]'
+  else
+    # Linux - using bash parameter expansion
+    echo "${1,,}"
+  fi
+}
```

Comment:
For the MacOS (zsh) support, I'd rather simplify the condition.

From: `if [[ -z "${IS_OPENSHIFT}" || "${IS_OPENSHIFT,,}" == "false" ]]; then`
To: `if [[ -z "${IS_OPENSHIFT}" || "${IS_OPENSHIFT}" == "false" ]]; then`

As you can see at [L1097](https://github.com/redhat-developer/rhdh/pull/2827/files#diff-5700e984b56368043ef9a0c1ab37e906444544c1fb8e47beb8189366fa0602c9L1097), the function that sets the `IS_OPENSHIFT` env. variable uses lower case, and it will not change throughout the test. It will not cause any issues.